packages/aws-cdk/test/integ/test-cli-backwards.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -euo pipefail
+integdir=$(cd $(dirname $0) && pwd)
+
+temp_dir=$(mktemp -d)
+
+function cleanup {
+    rm -rf ${temp_dir}
+    rm -rf ${integ_under_test}
+}
+
+function get_latest_published_version {
+
+    github_headers=""
+    if [[ "${GITHUB_TOKEN:-}" != "" ]]; then
+        github_headers="Authorization: token ${GITHUB_TOKEN}"
+    fi
+
+    out="${temp_dir}/aws-cdk-release.json"
+
+    curl -Ss --dump-header /dev/null -H "$github_headers" https://api.github.com/repos/aws/aws-cdk/releases/latest > ${out}
+    latest_version=$(node -p "require('${out}').name")
+    echo ${latest_version}
+}
+
+function download_repo {
+
+    version=$1
+
+    out="${temp_dir}/.repo.tar.gz"
+
+    curl -L -o ${out} "https://github.com/aws/aws-cdk/archive/${version}.tar.gz"
+    tar --strip-components=1 -zxf ${out} -C ${temp_dir}
+    echo ${temp_dir}
+
+}
+
+VERSION_UNDER_TEST=${VERSION_UNDER_TEST:-$(get_latest_published_version)}
+
+trap cleanup INT EXIT
+
+echo "Downloading aws-cdk repo version ${VERSION_UNDER_TEST}"
+temp_repo_dir="$(download_repo ${VERSION_UNDER_TEST})"
+
+# remove '/' that is prevelant in our branch names but causes
```

Comment:
where does the branch name come into play?

Reply:
It might be used as the "VERSION_UNDER_TEST", which is how i actually tested it.

Reply:
added comment in the code